third-party/linshare/src/test/java/org/apache/james/linshare/client/LinshareAPITest.java

```diff
@@ -50,48 +54,74 @@
 @ExtendWith(LinshareExtension.class)
 class LinshareAPITest {
 
+    @VisibleForTesting
     interface LinshareUserAPI {
 
         @RequestLine("GET /linshare/webservice/rest/user/v2/users")
         @Headers("Accept: application/json")
         List<User> allUsers();
     }
 
+    @VisibleForTesting
+    interface AuthorizationAPI {
+
+        static AuthorizationAPI from(String linshareUrl, UsernamePassword usernamePassword) throws Exception {
+            String base64Token = Base64.getEncoder()
+                .encodeToString((usernamePassword.getUsername() + ":" + usernamePassword.getPassword())
+                .getBytes(StandardCharsets.UTF_8));
+
+            LinshareConfiguration configuration = LinshareConfiguration.builder()
+                .urlAsString(linshareUrl)
+                .credential(LinshareCredential.basic(new LinshareCredential.Token(base64Token)))
```

Comment:
in order to retrieve jwt token, I still need to use basic authentication. Then, LinshareCredential should be generic enough to deal with Basic & JWT usages